src/Chapter3/ExamScore.hs

```diff
@@ -0,0 +1,106 @@
+module Chapter3.ExamScore2 where
+import Data.Char
+import Data.List
+     
+data StudentName = MkStudentName String deriving (Eq, Show, Ord)
+data SubjectName = MkSubjectName String deriving (Eq, Show, Ord)
+data SubjectMarks = MkSubjectMarks Int deriving (Eq, Show, Ord)
+data SubjectArr= MkSubjectArr [SubjectName] deriving (Eq, Show, Ord)
```

Comment:
how is `SubjectArr` providing any better semantics when compared to `[SubjectName]`?